test/instrumentation/graphql_test.rb

```diff
@@ -448,37 +444,21 @@ def company(id:)
       end
     end
 
-    describe 'loading' do
-    #   # in these 2 tests we are simulating the fact that the
-    #   # GraphQL::Tracing::AppOpticsTracing class
-    #   # from the graphql gem will be loaded first
-    #   it 'uses the newer version of AppOpticsTracing from the solarwinds_apm gem' do
-    #       load 'test/instrumentation/graphql/solarwinds_tracing_older.rb'
-    #       load 'lib/solarwinds_apm/inst/graphql.rb'
-    #
-    #       assert_match 'lib/solarwinds_apm/inst/graphql.rb',
-    #                    GraphQL::Tracing::AppOpticsTracing.new.method(:metadata).source_location[0]
-    #       assert_match 'lib/solarwinds_apm/inst/graphql.rb',
-    #                    GraphQL::Tracing::AppOpticsTracing.new.method(:platform_trace).source_location[0]
-    #   end
-    #
-    #   it 'uses the newer version of AppOpticsTracing from the graphql gem' do
-    #       load 'test/instrumentation/graphql/solarwinds_tracing_newer.rb'
-    #       load 'lib/solarwinds_apm/inst/graphql.rb'
-    #
-    #       assert_match 'graphql/solarwinds_tracing_newer.rb',
-    #                    GraphQL::Tracing::AppOpticsTracing.new.method(:metadata).source_location[0]
-    #       assert_match 'graphql/solarwinds_tracing_newer.rb',
-    #                    GraphQL::Tracing::AppOpticsTracing.new.method(:platform_trace).source_location[0]
-    #   end
-
-      it 'does not add plugins twice' do
-        GraphQL::Schema.use(GraphQL::Tracing::AppOpticsTracing)
-        GraphQL::Schema.use(GraphQL::Tracing::AppOpticsTracing)
-
-        assert_equal GraphQL::Schema.plugins.uniq.size, GraphQL::Schema.plugins.size,
-                     'failed: duplicate plugins found'
+    it 'does not add plugins twice' do
+      query_short = '{company (id: 1) { name}}'
+      SolarWindsAPM::SDK.start_trace('graphql_test') do
+        SolarWindsAPMTest::Schema.execute(query_short)
       end
+      size_1 = get_all_traces.size
+      clear_all_traces
+
+      SolarWindsAPMTest::Schema.use(GraphQL::Tracing::SolarWindsAPMTracing)
+      SolarWindsAPM::SDK.start_trace('graphql_test') do
+        SolarWindsAPMTest::Schema.execute(query_short)
+      end
+      size_2 = get_all_traces.size
+
+      assert_equal size_1, size_2
```

Comment:
Trivial: Could we could keep the `'failed: duplicate plugins found'` message to resemble the original test? Ultimately it probably doesn't matter so we can leave it like this!

Reply:
good point